addons/stock/models/stock_warehouse.py

```diff
@@ -395,6 +394,14 @@ def _create_or_update_route(self):
             rules_list = self._get_rule_values(
                 rules, values=route_data['rules_values'])
             # Create/Active rules
+            if route_data.get('last_rule_cancel') and rules_list and rules_list[-1].get('propagate_cancel'):
+                # in case of rule chain with cancel propagation. we need to stop
+                # the cancellation for the last step in order to avoid cancelling
+                # any other move after the chain.
+                # In the following flow:
+                # Input -> Quality check -> Stock -> Customer
+                # We want that cancelling I->GC cancel QC -> S but not S -> C
+                rules_list[-1]['propagate_cancel'] = False
```

Comment:
the comment is not 100% clear
I -> QC should have propagate_cancel to True
QC -> S should have propagate_cancel to False

Reply:
actually remove the comment since we'll adapt _get_rule_values